From 0c482060d599ea678db5ffa263ccbe0d1c4f1e2a Mon Sep 17 00:00:00 2001
From: Michael Brown <mcb30@ipxe.org>
Date: Thu, 17 Apr 2025 15:53:28 +0100
Subject: [PATCH] [undi] Work around broken ASUSTeK KNPA-U16 server PXE ROM

Signed-off-by: Michael Brown <mcb30@ipxe.org>
---
 src/arch/x86/drivers/net/undinet.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/arch/x86/drivers/net/undinet.c b/src/arch/x86/drivers/net/undinet.c
index 60e7ad799a..7fef9c49ca 100644
--- a/src/arch/x86/drivers/net/undinet.c
+++ b/src/arch/x86/drivers/net/undinet.c
@@ -861,6 +861,8 @@ static const struct undinet_irq_broken undinet_irq_broken_list[] = {
 	{ 0x8086, 0x1503, PCI_ANY_ID, PCI_ANY_ID },
 	/* HP 745 G3 laptop */
 	{ 0x14e4, 0x1687, PCI_ANY_ID, PCI_ANY_ID },
+	/* ASUSTeK KNPA-U16 server */
+	{ 0x8086, 0x1521, 0x1043, PCI_ANY_ID },
 };
 
 /**
